src/components/customplaybackrate/customplaybackrate.js

```diff
@@ -0,0 +1,91 @@
+import { playbackManager } from '../playback/playbackmanager';
+import template from './customplaybackrate.template.html';
+import './customplaybackrate.scss';
+
+let player;
+let rateInputField;
+let closeButton;
+let container;
+
+function init(instance) {
+    const parent = document.createElement('div');
+    document.body.appendChild(parent);
+    parent.innerHTML = template;
+
+    rateInputField = parent.querySelector('.customPlaybackRateInput');
+    closeButton = parent.querySelector('.customPlaybackRate-closeButton');
+    container = parent.querySelector('.customPlaybackRateContainer');
+
+    container.classList.add('hide');
+
+    rateInputField.value = '1.0';
+    rateInputField.hasFocus = false;
+
+    rateInputField.addEventListener('click', function () {
+        this.hasFocus = true;
+    });
+
+    rateInputField.addEventListener('keydown', function (event) {
+        if (event.key === 'Enter') {
+            const value = parseFloat(this.value);
+            if (!isNaN(value) && value >= 0.1 && value <= 10) {
+                playbackManager.setPlaybackRate(value, player);
+                CustomPlaybackRate.prototype.toggle('forceToHide');
+            } else {
+                alert('Enter a number between 0.1 and 10');
```

Comment:
You should not use native `alert` - use `alert` from `components/alert.js`:
https://github.com/jellyfin/jellyfin-web/blob/4e2b6290aaa4d2f99e0b190088b1bd0d89ee4ff8/src/apps/dashboard/controllers/encodingsettings.js#L6
It returns `Promise`, so you may need to adjust the code.

Also, the text of message should be translatable.